src/vmm/src/vmm_config/hotplug.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::io;
+
+use serde::{Deserialize, Serialize};
+
+use crate::vstate::vcpu::VcpuError;
+use crate::StartVcpusError;
+/// Unifying enum for all types of hotplug request configs.
+/// Currently only Vcpus may be hotplugged.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum HotplugRequestConfig {
+    /// Vcpu hotplug request
+    Vcpu(HotplugVcpuConfig),
+}
+
+/// Errors related to different types of hotplugging.
+/// Currently only Vcpus can be hotplugged.
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum HotplugRequestError {
+    /// Vcpu hotplugging error: {0}
+    Vcpu(#[from] HotplugVcpuError),
+}
+
+/// Errors associated with hot-plugging vCPUs
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum HotplugVcpuError {
+    /// The number of vCPUs added must be greater than 0.
+    VcpuCountTooLow,
+    /// The number of vCPUs added must be less than 32.
+    VcpuCountTooHigh,
+    /// Event fd error: {0}
+    EventFd(io::Error),
+    /// Error creating the vcpu: {0}
+    VcpuCreate(VcpuError),
+    /// Failed to start vCPUs
+    VcpuStart(StartVcpusError),
+}
+
+/// Config for hotplugging vCPUS
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct HotplugVcpuConfig {
+    /// Number of vcpus to start.
```

Comment:
This comment needs a bit of clarification. Does `vcpu_count` represent the amount of vCPUs the VM will end up with after (successfully) handling this request? If yes, then we might not "start" any vCPUs. We might actually shut down some.

Reply:
I see now that we only support adding new vCPUs. It would be nice to at least document that somewhere, ideally within the code itself. I would be in favour of having the request type look like this:

```rust
pub enum HotplugRequestConfig {
    /// Vcpu hotplug request
    AddVcpus(u8),
}
```

Reply:
At the moment, the value supplied is the number of vCPUs to be added. I'm not sure whether to change this to setting the total number of vCPUs, as this could be used to down VMs as well.

Reply:
Since using the `u8` thing doesn't work (see previous comment), this is fine. Resolving.